inst/include/dust/random/normal_ziggurat.hpp

```diff
@@ -28,6 +28,20 @@ real_type normal_ziggurat_tail(rng_state_type& rng_state, real_type x1,
   } while (true);
   return ret;
 }
+
+template <typename rng_state_type>
+int ziggurat_layer(rng_state_type& rng_state,
```

Comment:
I think `ziggurat_layer_draw` might be a little clearer as a name. Also a quick note to say what this is doing, as bitshifting is always a bit opaque

Reply:
Agreed on both counts, have added a bit of a wall of text now